pkg/pfcpsim/session/far_builder.go

```diff
@@ -0,0 +1,88 @@
+package session
+
+import (
+	"github.com/wmnsk/go-pfcp/ie"
+)
+
+type farBuilder struct {
+	farID       uint32
+	applyAction uint8
+	method      IEMethod
+}
+
+type downlinkFARBuilder struct {
+	farBuilder
+	teid       uint32
+	downlinkIP string
+}
+
+// NewFARBuilder returns a farBuilder containing initialized values to:
+// farID = 1, method = Create, applyAction = ActionDrop
+func NewFARBuilder() *farBuilder {
+	return &farBuilder{
+		farID:       uint32(1),
+		method:      Create,
+		applyAction: ActionDrop,
+	}
+}
+
+func (b *farBuilder) WithID(id uint32) *farBuilder {
+	b.farID = id
+	return b
+}
+
+func (b *farBuilder) WithMethod(method IEMethod) *farBuilder {
+	b.method = method
+	return b
+}
+
+func (b *farBuilder) WithAction(action uint8) *farBuilder {
+	b.applyAction = action
+	return b
+}
+
+func (b *farBuilder) BuildUplinkFAR() *ie.IE {
+	createFunc := ie.NewCreateFAR
```

Comment:
I believe we should have just a single `BuildFAR()`. The decision whether to build uplink/downlink FAR should be provided by user, e.g. by `MarkUplink()` or `MarkDownlink()`.

Reply:
Does this apply to `pdr_builder.go` as well?

Reply:
Yes